src/mono/browser/runtime/startup.ts

```diff
@@ -360,8 +360,9 @@ async function mono_wasm_init_threads() {
     if (!MonoWasmThreads) {
         return;
     }
-    const tid = getBrowserThreadID();
-    mono_set_thread_id(`0x${tid.toString(16)}-main`);
+    const tid = `0x${mono_wasm_main_thread_ptr().toString(16)}-main`;
+    mono_set_thread_id(tid);
```

Comment:
it seems like in some cases we use 'tid' to refer to a number and here we're using it to refer to a string. this should really be tname or threadName

Reply:
also to be clear i would love it if `mono_set_thread_id` could also be renamed.